### PR TITLE
Redpanda updates

### PR DESCRIPTION
**Pull request by simonl2002**
### Description

Please include a summary of the change and what type of change it is (new feature, bug fix, refactoring, documentation).
Please also include relevant motivation and context.
List any dependencies that are required for this change.

Fixes conduitio/conduit#1372

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit-connector-kafka/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,22 +26,22 @@ A source is getting associated with a consumer group ID the first time the `Read
...
-| `saslPassword`       | SASL password. If provided, a username needs to be provided too.                                                                                                                                             | false    |                           |
+| name                 | description                                                                                                                                                                                                  | required | default value                |
+|----------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|----------|------------------------------|
+| `servers`            | Servers is a list of Kafka bootstrap servers, which will be used to discover all the servers in a cluster.                                                                                                   | true     |                              |
```
Comment (hariso) [inline README.md:31]: There's a few Kafka references left in here.

File: connector.go [hunk 2]
```diff
@@ -12,9 +12,9 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-// Package kafka contains implementations for Kafka source and destination
+// Package repanda contains implementations for Redpanda source and destination
```
Comment (hariso) [inline connector.go:15]: typo in "repanda"

File: go.mod [hunk 3]
```diff
@@ -1,4 +1,4 @@
-module github.com/conduitio/conduit-connector-kafka
+module github.com/conduitio-labs/conduit-connector-redpanda
 
 go 1.21.1
```
Comment (hariso) [inline go.mod:3]: Maybe we can use the chance and update this?

File: README.md [hunk 4]
```diff
@@ -27,22 +26,22 @@ A source is getting associated with a consumer group ID the first time the `Read
...
+| `topics`             | Topics is a comma separated list of Kafka topics from which records will be read, ex: "topic1,topic2".                                                                                                       | true     |                              |
+| ~~`topic`~~          | Topic is the Kafka topic to read from. **Deprecated: use `topics` instead.**                                                                                                                                 | false    |                              |
```
Comment (raulb) [inline README.md:33]: This being a new connector, I think we should directly remove every reference of `topic` and leave `topics` exclusively.

File: README.md (hunk 1 above)
Comment (raulb) [reply README.md:31]: +1

File: connector.go (hunk 2 above)
Comment (raulb) [inline connector.go:15]:
```suggestion
// Package redpanda contains implementations for Redpanda source and destination
```

File: destination_integration_test.go [hunk 5]
```diff
@@ -12,15 +12,15 @@
 // See the License for the specific language governing permissions and
```
Comment (raulb) [inline destination_integration_test.go:12]: Theoretically, all copyrighted files should be @ 2024.

File: go.mod (hunk 3 above)
Comment (raulb) [reply go.mod:3]: +1 (go1.22.3 for reference)

File: README.md (hunk 4 above)
Comment (simonl2002) [reply README.md:33]: good point

File: go.mod (hunk 3 above)
Comment (simonl2002) [reply go.mod]: done

File: README.md (hunk 1 above)
Comment (simonl2002) [reply README.md]: done

File: README.md [hunk 6]
```diff
@@ -1,8 +1,7 @@
-# Conduit Connector Kafka
-![scarf pixel](https://static.scarf.sh/a.png?x-pxid=713ea3ba-66e0-4130-bdd0-d7db4b8706a0)
```
Comment (raulb) [inline README.md:2]: I seem to have lost to scarf so I can't corroborate where this one points out or how to create one if necessary (assuming it is) for redpanda. cc @simonl2002

**Review by raulb — APPROVED** [review]

**Pull request merged**